message/message-util/util/src/java/org/sakaiproject/message/util/BaseMessage.java

```diff
@@ -336,6 +344,28 @@ public String messageReference(String context, String channelId, String id)
 
 	} // messageReference
 
+	/**
+	 * Allows a service extending BaseMessageService to approve a message sender
+	 *
+	 * @param userId
+	 *        The user id of the message sender
+	 * @return true if the message sender is approved, false otherwise
+	 */
+	public boolean importAsDraft() {
+		return serverConfigurationService.getBoolean("import.importAsDraft", true);
+	}
+
+	/**
+	 * Allows a service extendingBase MessageService to approve a message sender
+	 *
+	 * @param userId
+	 *        The user id of the message sender
+	 * @return true if the message sender is approved, false otherwise
+	 */
+	public boolean approveMessageSender(String userId) {
+		return false;
```

Comment:
Is this false ok?

Reply:
Yes - Since this code is extended by both Announcements and MailArchive - messagesender logic only has meaning in the MailArchive case.   If you look at the default of this method in BaseMessage you will see the more complex logic.  I added these allbacks to avoid ugle if then else code in BaseMessage.